### PR TITLE
BL-1205, improve handling of locked files on download

### PR DESCRIPTION
**Pull request by JohnThomson**
- If we can't delete destination directory or move otherwise
fails, try copy.
- If we can't copy a particular file, see if it already exists with
the right content, and if so count this as success
- If we really can't make the destination directory correct,
copy all we can, and give a helpful message

### CODE REVIEW

File: src/BloomExe/WebLibraryIntegration/BloomS3Client.cs [hunk 1]
```diff
@@ -398,21 +440,47 @@ public string DownloadBook(string storageKeyOfBookFolder, string pathToDestinati
...
+
 				}
-				else
+				if (!done)
+					done = CopyDirectory(children[0], destinationPath);
+				if (!done)
 				{
-					CopyDirectory(children[0], destinationPath);
+					var msg = LocalizationManager.GetString("Download.CopyFailed",
+						"Bloom downloaded your book but had problems copying it to the bloom library. Possibly you already have a version of this book and some program is using it. You may need to restart your computer and download it again.");
```
Comment (hatton) [inline src/BloomExe/WebLibraryIntegration/BloomS3Client.cs]: "your" --> "the"
"copying it to the bloom library" --> "making it available in Bloom" 
"Possibly..." -> null
"You may need" --> "Please restart your computer and try again. If you get this message again, please write to us at issues@bloomlibrary.org".

File: src/BloomExe/WebLibraryIntegration/BloomS3Client.cs [hunk 2]
```diff
@@ -398,21 +440,47 @@ public string DownloadBook(string storageKeyOfBookFolder, string pathToDestinati
...
+					var caption = LocalizationManager.GetString("Download.Incomplete", "Download incomplete");
+					MessageBox.Show(null, msg, caption, MessageBoxButtons.OK, MessageBoxIcon.Warning);
```
Comment (hatton) [inline src/BloomExe/WebLibraryIntegration/BloomS3Client.cs]: How about a yellow box instead, so they can easily report it to us?

File: src/BloomExe/WebLibraryIntegration/BloomS3Client.cs (hunk 1 above)
Comment (JohnThomson) [inline src/BloomExe/WebLibraryIntegration/BloomS3Client.cs]: Done

File: src/BloomExe/WebLibraryIntegration/BloomS3Client.cs (hunk 2 above)
Comment (JohnThomson) [inline src/BloomExe/WebLibraryIntegration/BloomS3Client.cs]: Done.

**Pull request merged**